docs/src/dev/checklists.md

```diff
@@ -0,0 +1,157 @@
+
+### Row Sampler
+```
+1. Implementation
+- [] Create a file in the directory `linear_samplers`.
+- [] Create a mutable struct for the sampling method.
+- [] Create a constructor with default values for your struct.
+- [] Create a function `sample` (a template can be found in src/linear_samplers.jl):
+        INPUTS: 
+        1. [] `type`, your mutable struct
+        2. [] `A`, an `AbstractMatrix`
+        3. [] `b`, an `AbstractVector`
+        4. [] `x`, an `AbstractVector` 
+        5. []  `iter` an `Int64`
+        OUTPUTS:
+        1. [] `S` the sampling matrix (Note: if you are only sampling row indices this should be a vector of those indicies.)
+        2. [] `SA` the sampling matrix applied to A from the left.
+        3. [] `res` the sketched residual `SA * x - S * b`
+- [] Add an include("linear_samplers/[YOURFILE]") under the heading that best fits you method in src/linear_samplers.jl.
+- [] Add import statements to src/RLinearAlgebra.jl with any functions from other packages that you use.
+- [] Add your structure to the list of exported row samplers in src/RLinearAlgebra.jl.
+- [] Add your structure to docs/src/api/linear_samplers.md under the appropiate heading. 
+- [] Add a procedural test to test/linear_samplers.
+2. Pull request
+- [] Give a specific title to pull request.
+- [] Lay out the features added to the pull request.
+- [] Tag two people to review your pull request.
+```
+
+### Column Sampler
+```
+1. Implementation
+- [] Create a file in the directory `linear_samplers`.
+- [] Create a mutable struct for the sampling method.
+- [] Create a constructor with default values for your struct.
+- [] Create a function `sample` (a template can be found in src/linear_samplers.jl):
+        INPUTS: 
+        1. [] `type`, your mutable struct
+        2. [] `A`, an `AbstractMatrix`
+        3. [] `b`, an `AbstractVector`
+        4. [] `x`, an `AbstractVector` 
+        5. []  `iter` an `Int64`
+        OUTPUTS:
+        1. [] `S` the sampling matrix (Note: if you are only sampling column indices this should be a vector of those indicies.)
+        2. [] `AS` the sampling matrix applied to A from the right.
+        3. [] `res` the full residual `A * x - b` (Note the difference from the row methods).
+        4. [] `grad` the sketched residual `SA'res`.
+- [] Add an include("linear_samplers/[YOURFILE]") under the heading that best fits you method in src/linear_samplers.jl.
+- [] Add import statements to src/RLinearAlgebra.jl with any functions from other packages that you use.
+- [] Add your structure to the list of exported row samplers in src/RLinearAlgebra.jl.
+- [] Add your structure to docs/src/api/linear_samplers.md under the appropiate heading. 
+- [] Add a procedural test to test/linear_samplers.
+2. Pull request
+- [] Give a specific title to pull request.
+- [] Lay out the features added to the pull request.
+- [] Tag two people to review your pull request.
+```
```

Comment:
@nathanielpritchard column and row samplers are pretty much the same but the outputs change. I'd like to have a single set of instructions for a sampler (either row or column) but I realize that because the function return is different at the moment we cannot have that. 

No need to act on this for this PR but do you have any thoughts?

Reply:
Yeah I would ideally like to have a single checklist. I have left it as two to ensure that people remember to do both. In version 0.2 there won't be this distinction and therefore there will be only one checklist.